.github/workflows/ci.yaml

```diff
@@ -183,3 +182,72 @@ jobs:
           emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
           disable-animations: true
           script: ./gradlew connectedCheck -Pandroid.testInstrumentationRunnerArguments.class=org.ZingoLabs.Zingo.IntegrationTestSuite
+  
+  e2e_test:
+    name: End-to-End Test
+    runs-on: macos-12
+    strategy:
+      matrix:
+        include:
+          - arch : x86
+            api-level: 29
+            jniLibs_dir: x86
+          - arch : x86_64
+            api-level: 30
+            jniLibs_dir: x86_64
+    needs: [create-timestamp, build]
+    env:
+      timestamp: ${{ needs.create-timestamp.outputs.timestamp }}
+    steps:
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Gradle cache
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: 7.3.3
+          build-root-directory: ./android
+          cache-read-only: ${{ github.ref != 'refs/heads/main' && github.ref != 'refs/heads/dev' && github.ref != 'refs/pull/*' }}
```

Comment:
including github.ref != 'refs/pull/*' is good for devloping CI but shouldnt make it into dev. This is because if lots of PRs are open it will start overwriting the actual dev cache. I think either remove this or we could comment out this line with a comment saying "Only use this line for faster development, should not be merged into dev" and then have the standard one (only pushes to main and dev) uncommented

for ref https://github.com/zingolabs/zingo-mobile/blob/dev/.github/workflows/ci.yaml#L140